sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -74,39 +127,48 @@ class RapidsGdsStore(
     override def getMemoryBuffer: MemoryBuffer = getDeviceMemoryBuffer
 
     override def materializeMemoryBuffer: MemoryBuffer = {
-      val path = if (id.canShareDiskPaths) {
+      val path = if (id.canShareDiskPaths || isBatched(size)) {
         sharedBufferFiles.get(id)
       } else {
         id.getDiskPath(diskBlockManager)
       }
       closeOnExcept(DeviceMemoryBuffer.allocate(size)) { buffer =>
-        CuFile.readFileToDeviceBuffer(buffer, path, fileOffset)
-        logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+        if (cuFileBuffer.isEmpty) {
+          CuFile.readFileToDeviceBuffer(buffer, path, fileOffset)
+          logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+        } else {
+          buffer.copyFromMemoryBuffer(0, cuFileBuffer.get, fileOffset, size, Cuda.DEFAULT_STREAM)
+        }
         buffer
       }
     }
 
     override def copyToMemoryBuffer(srcOffset: Long, dst: MemoryBuffer, dstOffset: Long,
         length: Long, stream: Cuda.Stream): Unit = {
-      val path = if (id.canShareDiskPaths) {
+      val path = if (id.canShareDiskPaths || isBatched(size)) {
         sharedBufferFiles.get(id)
       } else {
         id.getDiskPath(diskBlockManager)
       }
       dst match {
         case dmOriginal: DeviceMemoryBuffer =>
           val dm = dmOriginal.slice(dstOffset, length)
-          // TODO: switch to async API when it's released, using the passed in CUDA stream.
-          CuFile.readFileToDeviceBuffer(dm, path, fileOffset + srcOffset)
-          logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+          if (cuFileBuffer.isEmpty) {
+            // TODO: switch to async API when it's released, using the passed in CUDA stream.
+            CuFile.readFileToDeviceBuffer(dm, path, fileOffset + srcOffset)
+            logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+          } else {
+            dm.copyFromMemoryBuffer(
+              0, cuFileBuffer.get, fileOffset + srcOffset, size, Cuda.DEFAULT_STREAM)
+          }
         case _ => throw new IllegalStateException(
           s"GDS can only copy to device buffer, not ${dst.getClass}")
       }
     }
 
     override protected def releaseResources(): Unit = {
       // Buffers that share paths must be cleaned up elsewhere
```

Comment:
This comment isn't addressed in this PR.  Currently when buffers share files there needs to be something external to the store that cleans up the files that are shared.  Before this, the only buffers sharing files were shuffle buffers, and the shuffle manager cleans up all the files when the shuffle session is garbage collected, so we don't accumulate files forever.  I don't see anything like that happening here, so it seems for a long-running GDS session there's nothing that will ever reap the files corresponding to batched buffers even if all the buffers in the batch have been destroyed.